https://issues.redhat.com/browse/ACM-12483 (#6676)
open-cluster-management/rhacm-docs: observability/customize_observability.adoc

[#customizing-certificates-object-store]
== Customizing certificates for accessing the object store

> To customize certificates for accessing the object store, complete the following steps:

Reviewer: Does customizing certificates make it easier to access the object store? Or does it enable users to access the object store? 

Reading your procedure steps, it seems like the benefit of access the object store is that it helps users make configurations for their specific development needs. So I added that here as well. Of course, omit if I'm incorrect :) 

If so, we could possibly rephrase to emphasize how customizing certificates benefit the user. What do you think about this rewrite? 

```suggestion
If you customize certificates, you can access the object store, giving you more options for configuration. To customize certificates, complete the following steps:
```

Author: How about the followng:
```suggestion
You can access the object store when you customize certificates, which give you more options to configure the observability service. Complete the following steps:
```

> . Create the secret with the `ca.crt` certificate. If you want to enable mutual TLS, you need to add the `public.crt` and `private.key` keys in the earlier secret. View the following example:

Reviewer: Is the example for the secrete they created? Or is the example for the key they created? 

To make it clearer, consider one of the following changes: 
```suggestion
. Create the secret with the `ca.crt` certificate.
.. If you want to enable mutual TLS, you need to add the `public.crt` and `private.key` keys in the earlier secret. View the following example:
```

Reviewer: ```suggestion
. Create the secret with the `ca.crt` certificate. View the following example:
.. *Optional:* If you want to enable mutual TLS, you need to add the `public.crt` and `private.key` keys in the earlier secret. 
```

Reviewer: Also, if you don't want it to be a sub step (..) I think you can make the the "key" step it's own step (.) . "If you want to enable..." implies the user doesn't _need_ to complete the step. So I don't think we need to make it a sub step.

Author: Its for the secret. Since the `public.crt` and `private.key` are not in the example I think it's best that we specify that it's optional and should come after the example

[#customizing-certificates-object-store]
== Customizing certificates for accessing the object store

To customize certificates for accessing the object store, complete the following steps:

> . Create the secret with the `ca.crt` certificate. If you want to enable mutual TLS, you need to add the `public.crt` and `private.key` keys in the earlier secret. View the following example:

Reviewer 2: I would extract the first sentence as the first step. This is the "secret creation" step.
And this is the second step: "reference the secret in the thanos object store config".
The last step below is: "reference the object store config in MCO"

Author: Not sure how my suggestion was removed: https://github.com/stolostron/rhacm-docs/pull/6676#discussion_r1688554750

+
[source,yaml]
----
 thanos.yaml: |
    type: s3
    config:
      ...
      http_config:
        tls_config:
          ca_file: /etc/minio/certs/ca.crt <1>
          cert_file: /etc/minio/certs/public.crt
          key_file: /etc/minio/certs/private.key
          insecure_skip_verify: false
----
+

> <1> The path to certificates and key values for the `thanos-object-storage` secret.

Reviewer 3: Can we mention "/etc/minio/certs/"  is the path defined in `tlsSecretMountPath` of the `MultiClusterObservability`.

. Configure the secret name and mount path by updating the `tlsSecretName` and `tlsSecretMountPath` parameters in the `MultiClusterObservability` custom resource. View the following example where the secret name is `tls-certs-secret` and the mount path for the certificates and key value is the directory that is used in the prior example:

+
[source,yaml]
----
metricObjectStorage:
      key: thanos.yaml
      name: thanos-object-storage
      tlsSecretName: tls-certs-secret
      tlsSecretMountPath: /etc/minio/certs
----

> . Update the `thanos.yaml` definition in the `thanos-object-storage` secret by adding the `http_config` section and add the certificate in the object store secret. View the following example:

Reviewer 2: We can remove everything up to line 471 with "To verify that you can access..." (keeping this)

Author: I'm not sure about removing the "Mount" step. There was a customer issue in the past where it was requested to add that.

Author: https://issues.redhat.com/browse/ACM-7716

Reviewer 2: From what I understand:

- `Update the `thanos.yaml` definition in the `thanos-object-storage`: this part is already done above.
- `Configure the secret name and mount path by updating the `tlsSecretName` and `tlsSecretMountPath` parameters in the `MultiClusterObservability``: this is already done above
- `Mount the secret in the `tlsSecretMountPath` resource of all components`: this step is invalid. The object store tls config defined in `metricObjectStorage` is automatically set in the components that need it. Plus those components don't have a field for specifying that.

But maybe I'm missing something.

Author: @thibaultmg excuse the delay, I have been working on doc build issues. I noticed that I did have a duplicate step. I see we are in different timezones, but I think a live review will be helpful. I will find time on your calendar so that we can get this resolved ASAP

Reviewer 4: I think there are some good conversation here on things that can be improved, but from what I see they are out of the scope of the ticket in question. So I wouldn't block the current changes.

>       tlsSecretMountPath: /etc/minio/certs
> ----
> 
> . Update the `thanos.yaml` definition in the `thanos-object-storage` secret by adding the `http_config` section and add the certificate in the object store secret. View the following example:

Author: Incorporating Brian's suggestion. However, I believe this is repetitive because we mention configuring this info in the resource in the previous step (Line 411), "Configure the secret name and mount path by updating the `tlsSecretName` and `tlsSecretMountPath` parameters in the `MultiClusterObservability` custom resource. View the following example where the secret name is `tls-certs-secret` and the mount path for the certificates and key value is the directory that is used in the prior example:"

```suggestion
      tlsSecretMountPath: /etc/minio/certs <1>
----
<1> The `/etc/minio/certs/` is the path defined in `tlsSecretMountPath` parameter of the `MultiClusterObservability` custom resource.

. Update the `thanos.yaml` definition in the `thanos-object-storage` secret by adding the `http_config` section and add the certificate in the object store secret. View the following example:
```

<1> Set the `insecure` parameter to `false` to enable HTTPS.
<2> The path for the `ca_file` parameter must match the `tlsSecretMountPath` from the `MultiClusterObservability` custom resource. The `ca.crt` must match the key in the `<tls_secret_name>` `Secret` resource.
+
*Optional:* If you want to enable mutual TLS, you need to add the `cert_file` and `key_file` keys to the `tls_config` section. See the following example:

+
[source,yaml]
----
 thanos.yaml: |
    type: s3
    config:
      bucket: "thanos"
      endpoint: "minio:9000"
      insecure: false
      access_key: "minio"
      secret_key: "minio123"
      http_config:
        tls_config:
          ca_file: /etc/minio/certs/ca.crt (1)
          cert_file: /etc/minio/certs/public.crt
          key_file: /etc/minio/certs/private.key
          insecure_skip_verify: false
----
<1> The path for `ca_file`, `cert_file`, and `key_file` must match the `tlsSecretMountPath` from the `MultiClusterObservability` custom resource. The `ca.crt`, `public.crt`, and `private.crt` must match the respective key in the `tls_secret_name>` `Secret` resource.

. To verify that you can access the object store, check that the pods are deployed.

> //what command can the user run to confirm that the pods are deployed | MJ | 08/08

Author: @douglascamata @moadz or @thibaultmg is there a command that the user can run to verify that the pods are deployed?

Reviewer 5: `oc -n open-cluster-management-observability get pods -l app.kubernetes.io/name=thanos-store`

>       tlsSecretMountPath: /etc/minio/certs
> ----
> 
> . Update the `thanos.yaml` definition in the `thanos-object-storage` secret by adding the `http_config` section and add the certificate in the object store secret. View the following example:

Author: Incorporating Brian's suggestion. However, I believe this is repetitive because we mention configuring this info in the resource in the previous step (Line 411), "Configure the secret name and mount path by updating the `tlsSecretName` and `tlsSecretMountPath` parameters in the `MultiClusterObservability` custom resource. View the following example where the secret name is `tls-certs-secret` and the mount path for the certificates and key value is the directory that is used in the prior example:"

```suggestion
      tlsSecretMountPath: /etc/minio/certs <1>
----
<1> The `/etc/minio/certs/` is the path defined in `tlsSecretMountPath` parameter of the `MultiClusterObservability` custom resource.

. Update the `thanos.yaml` definition in the `thanos-object-storage` secret by adding the `http_config` section and add the certificate in the object store secret. View the following example:
```
